### PR TITLE
refactor(utility): rename "LazyList" to "PagingList"

### PR DESCRIPTION
**Pull request by linjiX**
Make "PagingList" manage both lazy and non-lazy data.

- remove "LazyFactory.from_array"
- add "PagingList.from_factory"

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-290k588 将 LazyList 改名为 PagingList 统一管理本地数据和云端懒加载数据](https://app.clickup.com/t/290k588)

File: graviti/utility/lazy2.py [hunk 1]
```diff
@@ -100,29 +101,19 @@ def extend(self, length: int) -> None:
         self.total_count += length
 
 
-class LazyList:
-    """LazyList is a lazy load list.
+class PagingList:
+    """PagingList is a list combinated by multiple lists (pages).
```
Comment (Lee-000) [inline graviti/utility/lazy2.py:105]: Is `combinated` a correct word?
Use `composed of` instead?

File: graviti/utility/lazy2.py [hunk 2]
```diff
@@ -251,11 +242,38 @@ def _update_pages(self, start: int, stop: int, array: Optional[pa.Array] = None)
...
+            The PagingList instance created from given factory.
+
+        """
+        obj: _S = object.__new__(cls)
+        array_getter = partial(factory.get_array, keys=keys)
+        obj._pages = [
+            LazyPage(ranging, pos, array_getter)
+            for pos, ranging in enumerate(factory.get_page_ranges())
+        ]
+        # pylint: disable=protected-access
```
Comment (Lee-000) [inline graviti/utility/lazy2.py:266]: Why the disable comment is here?

File: graviti/utility/lazy2.py [hunk 3]
```diff
@@ -597,7 +555,7 @@ def get_array(self, pos: int, keys: Tuple[str, ...]) -> pa.Array:
 
         return array
 
-    def create_list(self, keys: Tuple[str, ...]) -> LazyList:
+    def create_list(self, keys: Tuple[str, ...]) -> PagingList:
         """Create a lazy list from the factory.
```
Comment (Lee-000) [inline graviti/utility/lazy2.py:559]: Should the docstring be updated too?

File: graviti/utility/lazy2.py [hunk 4]
```diff
@@ -26,6 +25,8 @@
 
 import pyarrow as pa
 
+_S = TypeVar("_S", bound="PagingList")
+
 
 class Offsets:
     """The offsets manager of the lazy list.
```
Comment (Lee-000) [inline graviti/utility/lazy2.py:32]: Should all the `lazy list` be updated?

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**